embrace-android-retrofit/src/main/java/io/embrace/android/embracesdk/retrofit/swazzle/callback/retrofit/EmbraceTracedConverterFactory.kt

```diff
@@ -0,0 +1,74 @@
+package io.embrace.android.embracesdk.retrofit.swazzle.callback.retrofit
+
+import io.embrace.android.embracesdk.Embrace
+import okhttp3.RequestBody
+import okhttp3.ResponseBody
+import retrofit2.Converter
+import retrofit2.Retrofit
+import java.lang.reflect.Type
+
+/**
+ * This class wraps a [Converter.Factory] and records spans around the
+ * serialization of retrofit request and response bodies.
+ */
+public class EmbraceTracedConverterFactory(
+    private val delegate: Converter.Factory
+) : Converter.Factory() {
+
+    private class ResponseBodyTracedConverter<T>(
+        private val delegate: Converter<ResponseBody, T>,
+        private val url: String
+    ) : Converter<ResponseBody, T> {
+
+        override fun convert(value: ResponseBody): T? {
+            var result: T? = null
+            Embrace.getInstance().internalInterface.recordSpan(
+                "Retrofit response body serialization",
```

Comment:
Trace names are used for aggregation, so lets use it's URL are part of the name for now?